--- a/VRage.EOS/VRage/EOS/MyEOSLobbyList.cs
+++ b/VRage.EOS/VRage/EOS/MyEOSLobbyList.cs
@@ -11,63 +11,59 @@ namespace VRage.EOS;
 
 internal class MyEOSLobbyList<T> where T : class
 {
-	[CompilerGenerated]
-	private sealed class _003C_003Ec__DisplayClass32_0
-	{
-		public MyEOSLobbyList<T> _003C_003E4__this;
+	public MyEOSLobbyList<T> _003C_003E4__this;
 
-		public List<T> lobbyList;
+	public List<T> lobbyList;
 
-		public List<string> idList;
+	public List<string> idList;
 
-		internal void _003CLoadServerListFromId_003Eb__0(Result x, (string ConnectingString, T Item)[] y)
+	internal void _003CLoadServerListFromId_003Eb__0(Result x, (string ConnectingString, T Item)[] y)
+	{
+		_003C_003E4__this.m_networking.InvokeOnMainThread(delegate
 		{
-			_003C_003E4__this.m_networking.InvokeOnMainThread(delegate
-			{
-				OnResults(x, y);
-			});
-			void Deliver()
+			OnResults(x, y);
+		});
+		void Deliver()
+		{
+			_003C_003E4__this.IsRequesting = false;
+			lock (_003C_003E4__this.m_lock)
 			{
-				_003C_003E4__this.IsRequesting = false;
-				lock (_003C_003E4__this.m_lock)
+				_003C_003E4__this.m_lobbyIds = idList;
+				_003C_003E4__this.m_lobbyList = lobbyList;
+				(string, T, bool) result;
+				while (_003C_003E4__this.m_pendingOps.TryDequeue<(string, T, bool)>(out result))
 				{
-					_003C_003E4__this.m_lobbyIds = idList;
-					_003C_003E4__this.m_lobbyList = lobbyList;
-					(string, T, bool) result;
-					while (_003C_003E4__this.m_pendingOps.TryDequeue<(string, T, bool)>(out result))
+					if (result.Item3)
+					{
+						_003C_003E4__this.AddServer(result.Item1, result.Item2);
+					}
+					else
 					{
-						if (result.Item3)
-						{
-							_003C_003E4__this.AddServer(result.Item1, result.Item2);
-						}
-						else
-						{
-							_003C_003E4__this.RemoveServer(result.Item1, result.Item2);
-						}
+						_003C_003E4__this.RemoveServer(result.Item1, result.Item2);
 					}
 				}
-				_003C_003E4__this.Log($"Server list queried: {lobbyList.Count} result(s).");
-				_003C_003E4__this.m_queryCompletion?.Invoke();
 			}
-			void OnResults(Result result, (string ConnectingString, T Item)[] servers)
+			_003C_003E4__this.Log($"Server list queried: {lobbyList.Count} result(s).");
+			_003C_003E4__this.m_queryCompletion?.Invoke();
+		}
+		void OnResults(Result result, (string ConnectingString, T Item)[] servers)
+		{
+			if (result != Result.Success)
+			{
+				_003C_003E4__this.m_networking.Error($"Lobby search was not successful: {result}");
+			}
+			for (int i = 0; i < servers.Length; i++)
 			{
-				if (result != Result.Success)
+				if (_003C_003E4__this.m_networking.VerboseLogging)
 				{
-					_003C_003E4__this.m_networking.Error($"Lobby search was not successful: {result}");
+					_003C_003E4__this.Log("Server [" + servers[i].ConnectingString + "]: " + ((servers[i].Item == null) ? "Not Available" : "Loaded") + ".");
 				}
-				for (int i = 0; i < servers.Length; i++)
+				if (servers[i].Item != null)
 				{
-					if (_003C_003E4__this.m_networking.VerboseLogging)
-					{
-						_003C_003E4__this.Log("Server [" + servers[i].ConnectingString + "]: " + ((servers[i].Item == null) ? "Not Available" : "Loaded") + ".");
-					}
-					if (servers[i].Item != null)
-					{
-						lobbyList.Add(servers[i].Item);
-					}
+					lobbyList.Add(servers[i].Item);
 				}
-				Deliver();
 			}
+			Deliver();
 		}
 	}
 
--- a/VRage.EOS/VRage/EOS/MyEOSLobbyList.cs
+++ b/VRage.EOS/VRage/EOS/MyEOSLobbyList.cs
@@ -433,17 +415,17 @@ internal class MyEOSLobbyList<T> where T : class
 		{
 			if (result != Result.Success)
 			{
-				((_003C_003Ec__DisplayClass32_0)this)._003C_003E4__this.m_networking.Error($"Lobby search was not successful: {result}");
+				this.m_networking.Error($"Lobby search was not successful: {result}");
 			}
 			for (int i = 0; i < servers.Length; i++)
 			{
-				if (((_003C_003Ec__DisplayClass32_0)this)._003C_003E4__this.m_networking.VerboseLogging)
+				if (this.m_networking.VerboseLogging)
 				{
-					((_003C_003Ec__DisplayClass32_0)this)._003C_003E4__this.Log("Server [" + servers[i].ConnectingString + "]: " + ((servers[i].Item == null) ? "Not Available" : "Loaded") + ".");
+					this.Log("Server [" + servers[i].ConnectingString + "]: " + ((servers[i].Item == null) ? "Not Available" : "Loaded") + ".");
 				}
 				if (servers[i].Item != null)
 				{
-					((_003C_003Ec__DisplayClass32_0)this).lobbyList.Add(servers[i].Item);
+					this.lobbyList.Add(servers[i].Item);
 				}
 			}
 			Deliver();
