--- a/VRage.Game/VRage/Game/ObjectBuilders/Components/MyObjectBuilder_HierarchyComponentBase.cs
+++ b/VRage.Game/VRage/Game/ObjectBuilders/Components/MyObjectBuilder_HierarchyComponentBase.cs
@@ -16,7 +16,7 @@ namespace VRage.Game.ObjectBuilders.Components;
 public class MyObjectBuilder_HierarchyComponentBase : MyObjectBuilder_ComponentBase
 {
 
-	[ProtoMember(1, AsReference = true)]
+	[ProtoMember(1)]
 	[XmlArrayItem("MyObjectBuilder_EntityBase", Type = typeof(MyAbstractXmlSerializer<MyObjectBuilder_EntityBase>))]
 	[DynamicItem(typeof(MyObjectBuilderDynamicSerializer), true)]
 	public List<MyObjectBuilder_EntityBase> Children = new List<MyObjectBuilder_EntityBase>();
